runtime/gc_glue_java/MetronomeDelegate.cpp

```diff
@@ -1629,15 +1629,22 @@ stackSlotIteratorForRealtimeGC(J9JavaVM *javaVM, J9Object **slotPtr, void *local
 	if (realtimeMarkingScheme->isHeapObject(object)) {
 		/* heap object - validate and mark */
 		Assert_MM_validStackSlot(MM_StackSlotValidator(0, object, stackLocation, walkState).validate(env));
-		realtimeMarkingScheme->markObject(env, object);
+		if (MUTATOR_THREAD == env->getThreadType()) {
```

Comment:
It would be slightly more efficient to move this check before calling stack slot iterator (so doing it only once per whole stack, rather then doing it per each slot). 
It would require to specialize the callbacks (stackSlotIteratorForRealtimeGCFromMutator/FromGC). But, for now, I'll assume these stacks are small, and prefer code sharing, instead.